### PR TITLE
update user_profile, refactor userStore,add loader to the timeline and  fix the null value of the module and TimeLine  component

### PR DESCRIPTION
**Pull request by yousefallaban**

### CODE REVIEW

**Comment by yousefallaban** [comment]
[change header size README.md](https://trello.com/c/QCKuox3X/1-change-header-size-readmemd)

**Comment by malnajar** [comment]
[Apply the loader on timeline](https://trello.com/c/XeGRiCRD/9-apply-the-loader-on-timeline)

**Comment by malnajar** [comment]
[Apply the loader on timeline](https://trello.com/c/XeGRiCRD/9-apply-the-loader-on-timeline)

**Comment by yousefallaban** [comment]
[add loader to the timeline, format the Doc and fix the null value of the module component](https://trello.com/c/vmnc3HAe/10-add-loader-to-the-timeline-format-the-doc-and-fix-the-null-value-of-the-module-component)

**Comment by yousefallaban** [comment]
[add loader to the timeline, format the Doc and fix the null value of the module component](https://trello.com/c/vmnc3HAe/10-add-loader-to-the-timeline-format-the-doc-and-fix-the-null-value-of-the-module-component)

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,2 +1,6 @@
-{"javascript.implicitProjectConfig.experimentalDecorators": true,
+{
```
Comment (M3kH) [inline .vscode/settings.json]: This is not related with the asked changed.

Comment (M3kH) [reply .vscode/settings.json]: do a `git fetch && git checkout origin/master -- .vscode/settings.json`

File: src/Pages/Modules/ModuleFooter.js [hunk 2]
```diff
@@ -45,25 +51,41 @@ export default class ModuleFooter extends Component {
...
   hideAddModal = () => {
     this.setState({
-      isadding: false
+      isAdding: false
     });
   }
-  
+
   render() {
+    console.log('fromMF')
```
Comment (M3kH) [inline src/Pages/Modules/ModuleFooter.js]: No console.log

File: src/Pages/Users/Users.js [hunk 3]
```diff
@@ -77,25 +36,25 @@ export default class Users extends React.Component {
           className={styles.userSearchBox}
           type="text"
           placeholder="lookup someone"
-          onChange={this.searchUser}
+          onChange={store.searchUser}
```
[the remark this answers is not in the document]
Comment (M3kH) [reply src/Pages/Users/Users.js]: https://twitter.com/ideabile/status/986097938828013568

File: src/store/UserStore.js [hunk 4]
```diff
@@ -34,36 +32,123 @@ class store {
...
 		}
 	}
 
-	notify() {
-		for (const handler of this.handlers) {
-			handler(this.state)
+
+	loadUsers = () => {
+		// console.log(this.state.users[1])
+		fetch('http://localhost:3000/api/users', {
```
Comment (M3kH) [inline src/store/UserStore.js]: This url is `flaky` => https://www.vocabulary.com/dictionary/flaky

Works locally not in the internet 🕸

File: src/store/UserStore.js [hunk 5]
```diff
@@ -34,36 +32,123 @@ class store {
...
 			}
 		}
 	}
 
-	notify() {
-		for (const handler of this.handlers) {
-			handler(this.state)
+
+	loadUsers = () => {
+		// console.log(this.state.users[1])
```
Comment (M3kH) [inline src/store/UserStore.js]: No console.lo

**Comment by M3kH** [comment]
Keep doing a good job, fetch and resolve the conflict!

File: src/Pages/Users/Users.js [hunk 6]
```diff
@@ -26,49 +23,11 @@ export default class Users extends React.Component {
   }
 
   componentDidMount = () => {
-    this.loadUsers();
+    store.loadUsers();
```
[the remark this answers is not in the document]
Comment (yousefallaban) [reply src/Pages/Users/Users.js]: I move the loadUsers() function from the child component to the store, so we don't need  this any more, and yes it set the state.

File: src/Pages/Users/Users.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (yousefallaban) [reply src/Pages/Users/Users.js]: here it's the same, best practices to be the function where the state live, and move it  to the parent component. so the stat in the store and the function should be in the same level, that what i come up with

**Comment by M3kH** [comment]
Good job @yousefallaban :-)
